code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -96,9 +99,21 @@
 /obj/effect/alien/resin/attackby(obj/item/W, mob/user)
 	if(!(W.flags_item & NOBLUDGEON))
 		var/damage = W.force
-		if(W.w_class < 4 || !W.sharp || W.force < 20) //only big strong sharp weapon are adequate
-			damage /= 4
-		health -= damage
+		var/multiplier = 1
+		if(W.damtype == "burn") //Burn damage deals extra vs resin structures (mostly welders).
+			multiplier += 1
+			if(istype(W, /obj/item/tool/pickaxe/plasmacutter))
+				var/obj/item/tool/pickaxe/plasmacutter/P = W
+				if(P.cell.charge >= P.charge_cost * PLASMACUTTER_VLOW_MOD)
+					multiplier += PLASMACUTTER_RESIN_MULTIPLIER //Plasma cutters are particularly good at destroying resin structures.
+					P.cut_apart(user, src.name, src, P.charge_cost * PLASMACUTTER_VLOW_MOD) //Minimal energy cost.
+				else
+					P.fizzle_message(user)
+					return
+		else if((W.w_class < 4 || !W.sharp || W.force < 20) && !W.damtype == "burn" ) //only big strong sharp weapon are adequate
```

Comment:
`!W.damtype == "burn"` is not necessary as there's an `else` already doing that.
On the rest, no need to buff the bayonet weapon against resins. please fix this.

Reply:
This is actually default.

Reply:
Doesn't mean we need to keep it.